apps/ui/src/components/pages/application/utils/validators.ts

```diff
@@ -67,6 +75,16 @@ export const VerifySectionsTouched = (fields?: ApplicationContentsResponse) => {
 				...sectionTouched,
 				institutional: true,
 			};
+		} else if (isAgreementKey(key) && value !== null) {
+			sectionTouched = {
+				...sectionTouched,
+				agreement: true,
+			};
+		} else if (isAppendicesKey(key) && value !== null) {
```

Comment:
`forEach` could be probably reorganized so `&& value !== null` isn't repeated in each conditional? One `value === null` early on?